aiida/tools/importexport/common/archive.py

```diff
@@ -218,9 +222,70 @@ def _read_json_file(self, filename):
             return json.load(fhandle)
 
 
-def extract_zip(infile, folder, nodes_export_subfolder=None, silent=False):
+def update_description(path, refresh: bool = False):
+    """Update description for a progress bar given path
+
+    :param path: path of file or directory
+    :type path: str
+    :param refresh: Whether or not to refresh the progress bar with the new description. Default: False.
+    :type refresh: bool
     """
-    Extract the nodes to be imported from a zip file.
+    (path, description) = os.path.split(path)
+    while description == '':
+        (path, description) = os.path.split(path)
+    description = 'EXTRACTING: {}'.format(description)
+
+    progress_bar = get_progress_bar()
+    progress_bar.set_description_str(description, refresh=refresh)
+
+
+def get_file_iterator(file_handle, folderpath, silent=False, **kwargs):  # pylint: disable=unused-argument
+    """Go through JSON files and then return new file_iterator
+
+    :param file_handle: A file handle returned from `with open() as file_handle:`.
+    :type file_handle: `tarfile.TarFile`, `zipfile.ZipFile`
+
+    :param folderpath: Path to folder.
+    :type folderpath: str
+
+    :param silent: suppress progress bar.
+    :type silent: bool
+
+    :return: List of filenames in the archive.
+    :rtype: list
```

Comment:
Is this correct? I can only see `return get_progress_bar` and `get_progress_bar` should return an instance of progress bar, not a list?

Reply:
True. This is a failure to update the doc string after the last big re-write.
But as information: The `tqdm` progress bar can act as a replacement for the iterable one wants to traverse, at the same time updating the progress bar for each run of the traversal. So it's more helpful to think of it here as an iterable. However, the instance that is returned, is indeed still `tqdm.tqdm`.